zigen-project/zen
# show wayland client's popup/subsurface

## Context



https://github.com/zigen-project/zen/issues/116



## Summary



In this PR, I focused on showing/hiding popup and subsurface.

- [x] show popup surface
- [x] hide popup surface
- [x] show subsurface
- [x] hide subsurface

Another features related to popup/subsurface yet not implemented are described in issue #116 #110  and it will be implemented soon.

## How to check behavior

1. run zen-desktop with a wayland client such as weston-terminal.
2. right-click on the wayland client.
3. popup appears near cussor.
4. click another area, and the popup disappears.
